pkg/scheduler/algorithm_factory.go

```diff
@@ -51,45 +51,68 @@ type AlgorithmFactoryArgs struct {
 // PriorityMetadataProducerFactory produces MetadataProducer from the given args.
 type PriorityMetadataProducerFactory func(AlgorithmFactoryArgs) priorities.MetadataProducer
 
-// FitPredicateFactory produces a FitPredicate from the given args.
-type FitPredicateFactory func(AlgorithmFactoryArgs) predicates.FitPredicate
-
-// PriorityFunctionFactory produces map & reduce priority functions
-// from a given args.
-type PriorityFunctionFactory func(AlgorithmFactoryArgs) (priorities.PriorityMapFunction, priorities.PriorityReduceFunction)
-
-// PriorityConfigFactory produces a PriorityConfig from the given function and weight
-type PriorityConfigFactory struct {
-	MapReduceFunction PriorityFunctionFactory
-	Weight            int64
-}
-
 var (
-	schedulerFactoryMutex sync.RWMutex
-
-	// maps that hold registered algorithm types
-	// TODO(Huang-Wei): remove this.
-	fitPredicateMap        = make(map[string]FitPredicateFactory)
-	mandatoryFitPredicates = sets.NewString()
-	priorityFunctionMap    = make(map[string]PriorityConfigFactory)
-	algorithmProviderMap   = make(map[string]AlgorithmProviderConfig)
+	// predicate keys supported for backward compatibility with v1.Policy.
```

Comment:
NOTE: the following maps replace the explicit registration in the register_predicates.go and register_priorities.go. This is temporary, a follow up PR will remove those and directly use ConfigProducerRegistry defined in framework/plugins